### PR TITLE
[#7325] Split test_icommands_file_operations.py contents across multiple classes (main)

### PR DESCRIPTION
**Pull request by korydraughn**
This PR splits a python unit test classes into 5 classes.

All tests pass.

I'm going to see how much time each group of tests take so we know if tests need to be moved around.

I'm hoping we can get each group of tests to complete within 10 to 15 minutes.

### CODE REVIEW

File: scripts/irods/test/test_icommands_file_operations.py [hunk 1]
```diff
@@ -535,6 +555,23 @@ def test_irsync_r_nested_coll_to_coll(self):
...
                             "Extra files in vault:\n" + str(files_in_vault - local_files))
 
+@unittest.skipIf(test.settings.TOPOLOGY_FROM_RESOURCE_SERVER, "Skip for topology testing from resource server")
+class Test_ICommands_File_Operations_3(resource_suite.ResourceBase, unittest.TestCase):
+    plugin_name = IrodsConfig().default_rule_engine_plugin
+    class_name = 'Test_ICommands_File_Operations_3'
+
+    def setUp(self):
+        # TODO: Remove this sleep when #4359 is resolved
+        time.sleep(30)
```
Comment (trel) [inline scripts/irods/test/test_icommands_file_operations.py:565]: note that https://github.com/irods/irods/issues/4359 is closed.    investigate removing this sleep (and others in this file).

Comment (korydraughn) [reply scripts/irods/test/test_icommands_file_operations.py:565]: Yes. That was part of the plan too.

I'm going to address that in another commit or separate PR.

**Review by alanking — COMMENTED** [review]
I'm guessing that this is a temporary measure until we can reach a more proper solution?

**Comment by korydraughn** [comment]
Sure. I guess it can be viewed that way.

I'm assuming you're referring to a future where we can remove the core_tests_list.json?

**Comment by alanking** [comment]
I am more referring to the removal of `test_icommands_file_operations.py`. We need to split this up into more targeted files. I don't think we should expend that effort for 4.3.1.

**Comment by korydraughn** [comment]
I see. In that case, yes, this can viewed as a temporary improvement.

And yes, a post 4.3.1 type of change for sure.

**Comment by SwooshyCueb** [comment]
Do all of these classes really need to inherit from `ResourceBase`? My understanding is that this introduces a lot more overhead than the sessions mixin (see `test_itouch.py` for usage).

**Comment by alanking** [comment]
> Do all of these classes really need to inherit from `ResourceBase`? My understanding is that this introduces a lot more overhead than the sessions mixin (see `test_itouch.py` for usage).

It is likely that many of the tests are relying on things set up by `ResourceBase`, but I guarantee that it's not all of them.

**Comment by korydraughn** [comment]
Right. This is the simplest tweak I can make without having to investigate several things.

Dropping that base class can happen after 4.3.1 is released.

File: scripts/irods/test/test_icommands_file_operations.py (hunk 1 above)
Comment (korydraughn) [reply scripts/irods/test/test_icommands_file_operations.py:565]: Removing the call to sleep did not break anything. All of the tests passed.

**Comment by korydraughn** [comment]
Based on the following times, test suite 2 takes significantly longer to complete than the rest. That is likely due to the number of test files being created (i.e. the majority of the tests which deal with recursive puts, etc are in test suite 2.
```
2023-09-19 12:16:08,249 - ==== begin test run results ====
-----
results for [ubuntu-2004-postgres-1012_irods-catalog-provider_1]
        passed tests:
                [[ 417.3485]s]  [test_icommands_file_operations.Test_ICommands_File_Operations_1]
        skipped tests:
        failed tests:
        return code:[0]
        time elapsed: [    417.3]seconds ([   0]hours [    6.956]minutes)
-----
-----
results for [ubuntu-2004-postgres-1012_irods-catalog-provider_2]
        passed tests:
                [[1540.0460]s]  [test_icommands_file_operations.Test_ICommands_File_Operations_2]
        skipped tests:
        failed tests:
        return code:[0]
        time elapsed: [ 1.54e+03]seconds ([   0]hours [    25.67]minutes)
-----
-----
results for [ubuntu-2004-postgres-1012_irods-catalog-provider_3]
        passed tests:
                [[ 498.4376]s]  [test_icommands_file_operations.Test_ICommands_File_Operations_3]
        skipped tests:
        failed tests:
        return code:[0]
        time elapsed: [    498.4]seconds ([   0]hours [    8.307]minutes)
-----
-----
results for [ubuntu-2004-postgres-1012_irods-catalog-provider_4]
        passed tests:
                [[ 133.2004]s]  [test_icommands_file_operations.Test_ICommands_File_Operations_4]
        skipped tests:
        failed tests:
        return code:[0]
        time elapsed: [    133.2]seconds ([   0]hours [     2.22]minutes)
-----
-----
results for [ubuntu-2004-postgres-1012_irods-catalog-provider_5]
        passed tests:
                [[ 120.3312]s]  [test_icommands_file_operations.Test_ICommands_File_Operations_5]
        skipped tests:
        failed tests:
        return code:[0]
        time elapsed: [    120.3]seconds ([   0]hours [    2.006]minutes)
-----
All tests passed! :)
time elapsed: [1540.0518]seconds ([   0]hours [25.6675]minutes)
==== end of test run results ====

2023-09-19 12:16:08,383 - version:[4.3.0]
2023-09-19 12:16:08,383 - sha:[8dbd9477ab96e08ba051888c17bf311cd5e2479b]
```
I'm going to move some tests around to distribute the load better.

File: scripts/irods/test/test_icommands_file_operations.py (hunk 1 above)
Comment (trel) [reply scripts/irods/test/test_icommands_file_operations.py:565]: glory be.

**Comment by korydraughn** [comment]
After moving a few tests around, we're now looking at the following.
```
2023-09-19 13:10:54,719 - ==== begin test run results ====
-----
results for [ubuntu-2004-postgres-1012_irods-catalog-provider_1]
        passed tests:
                [[ 409.0563]s]  [test_icommands_file_operations.Test_ICommands_File_Operations_1]
        skipped tests:
        failed tests:
        return code:[0]
        time elapsed: [    409.1]seconds ([   0]hours [    6.818]minutes)
-----
-----
results for [ubuntu-2004-postgres-1012_irods-catalog-provider_2]
        passed tests:
                [[ 594.5719]s]  [test_icommands_file_operations.Test_ICommands_File_Operations_2]
        skipped tests:
        failed tests:
        return code:[0]
        time elapsed: [    594.6]seconds ([   0]hours [     9.91]minutes)
-----
-----
results for [ubuntu-2004-postgres-1012_irods-catalog-provider_3]
        passed tests:
                [[ 498.0229]s]  [test_icommands_file_operations.Test_ICommands_File_Operations_3]
        skipped tests:
        failed tests:
        return code:[0]
        time elapsed: [    498.0]seconds ([   0]hours [      8.3]minutes)
-----
-----
results for [ubuntu-2004-postgres-1012_irods-catalog-provider_4]
        passed tests:
                [[ 599.9745]s]  [test_icommands_file_operations.Test_ICommands_File_Operations_4]
        skipped tests:
        failed tests:
        return code:[0]
        time elapsed: [    600.0]seconds ([   0]hours [     10.0]minutes)
-----
-----
results for [ubuntu-2004-postgres-1012_irods-catalog-provider_5]
        passed tests:
                [[ 553.5324]s]  [test_icommands_file_operations.Test_ICommands_File_Operations_5]
        skipped tests:
        failed tests:
        return code:[0]
        time elapsed: [    553.5]seconds ([   0]hours [    9.226]minutes)
-----
All tests passed! :)
time elapsed: [ 599.9881]seconds ([   0]hours [ 9.9998]minutes)
==== end of test run results ====

2023-09-19 13:10:54,882 - version:[4.3.0]
2023-09-19 13:10:54,882 - sha:[90cfd221755d2c08717c1555e77c2345e51b1f86]
```
I think this is in a good spot. Pushing changes now.

**Comment by trel** [comment]
excellent.   does this 8-10min window appear to be similar to other test suites in the project?

**Comment by korydraughn** [comment]
No. Many tests finish much sooner than 8-10 minutes while others can take up to 15+ minutes. The 15+ minute test suites are typically the resource tests. The slowness of those tests is likely due to the setup and tear-down around each test case.

**Comment by korydraughn** [comment]
This PR is complete now.

Once approved, I'll add the pounds.

**Comment by trel** [comment]
> The slowness of those tests is likely due to the setup and tear-down around each test case.

makes sense.   a later effort to optimize once we can run things remotely against a cluster.

**Review by trel — APPROVED** [review]
approach looks good - if this is not changing the actual tests.. just chopping/splitting and moving them around... i approve.

**Comment by korydraughn** [comment]
Right.

**Comment by korydraughn** [comment]
There are no changes to the tests. This is just dividing up the work to take advantage of the testing environments parallelism features.

**Comment by korydraughn** [comment]
Are we ready for pounds?

**Comment by trel** [comment]
i think so.   these will be run many times before release.   go for it.

**Comment by korydraughn** [comment]
Added #'s

**Review by alanking — APPROVED** [review]
Here we gooo

**Pull request merged**